# Grundbegriffe 

Beantworte und diskutiere die folgenden Fragen konkret für den in Einheit 1 heruntergeladenen San Francisco Library Usage Datensatz. Notiere Deine Ergebnisse in Stichpunkte.

In [27]:
import pandas as pd

In [28]:
df = pd.read_csv("/home/katja/2024-2025-Data_Lirbrarian_Katja_Gödde/Modul_3/data/Library_Usage.csv", dtype={"Within San Francisco County": str}, na_values=['Null', 'NA', ''])  

In [29]:
df

Patron Type Definition  Total Checkouts  Total Renewals  \
0                      Senior                5               0   
1                       Adult                0               0   
2                       Adult                0               0   
3                     Welcome                1               1   
4                      Senior                0               0   
...                       ...              ...             ...   
450354    Digital Access Card                0               0   
450355    Digital Access Card                0               0   
450356    Digital Access Card                0               0   
450357    Digital Access Card                0               0   
450358    Digital Access Card                0               0   

                Age Range Home Library Definition Circulation Active Month  \
0       75 years and over                    Main                      Nov   
1          45 to 54 years                    Main                      Jul   
2          55 to 59 years        Western Addition                      Mar   
3          20 to 24 years                Richmond                      Aug   
4          65 to 74 years                  Sunset                      Mar   
...                   ...                     ...                      ...   
450354     35 to 44 years               Ingleside                      NaN   
450355     45 to 54 years               Ingleside                      Aug   
450356     25 to 34 years               Ingleside                      Aug   
450357     35 to 44 years               Ingleside                      Apr   
450358     65 to 74 years                 Bayview                      Apr   

        Circulation Active Year Notice Preference Definition  \
0                        2022.0                        Email   
1                        2023.0                        Email   
2                        2024.0                        Email   
3                        2022.0                        Email   
4                        2024.0                        Print   
...                         ...                          ...   
450354                      NaN                        Email   
450355                   2022.0                        Print   
450356                   2022.0                        Print   
450357                   2022.0                        Print   
450358                   2023.0                        Email   

        Provided Email Address  Year Patron Registered  \
0                         True                    2015   
1                         True                    2019   
2                         True                    2022   
3                         True                    2022   
4                        False                    2023   
...                        ...                     ...   
450354                   False                    2023   
450355                   False                    2022   
450356                   False                    2022   
450357                   False                    2022   
450358                   False                    2023   

       Within San Francisco County  
0                            False  
1                            False  
2                            False  
3                            False  
4                            False  
...                            ...  
450354                         NaN  
450355                         NaN  
450356                         NaN  
450357                         NaN  
450358                         NaN  

[450359 rows x 11 columns]

Wie viele Merkmale besitzt der Datensatz?
Der Datensatz besitzt 11 Merkmale, die hier durch die Spaltennamen repräsentiert werden.

Wie groß ist die Stichprobengröße des Datensatzes? 450359 (= die Anzahl der Zeilen)

Wer oder was sind die Merkmalsträger? Die Merkmalsträger sind die Bibliotheksnutzer*innen an denen die Ausprägungen gemessen werden.

Von wann bis wann wurden die Daten erhoben? Vom 02. Dezember 2016 bis zum 25. März 2024 (einsehbar in den Metadaten unter https://data.sfgov.org/Culture-and-Recreation/Library-Usage/qzz6-2jup/about_data)

Wie lässt sich die Grundgesamtheit beschreiben? Handelt es sich um eine Vollerhebung? Es handelt sich nicht um eine Vollerhebung, da nicht alle Kunden, sondern nur aktive Kunden mit einbezogen wurden. Aktive Kunden sind Kunden mit noch nicht abgelaufenen Bibliotheksausweisen und Kunden, die innerhalb der letzten drei Jahre eine Ausleihtätigkeit ausgeübt haben.

Welche Merkmale sind stetig? Welche diskret?
- Patron Type Definition: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Total Checkouts: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Total Reneweals: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Age Range: Diskret, da es sich um abzählbar viele Gruppierungen von Altersklassen handelt. Wenn es keine Gruppierung gäbe, wäre das Merkmal dagegen stetig.
- Home Library Definition: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Circulation Active Month: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Circulation Active Year: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Notice Preference Definition: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Provided Email Address: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Year Patron Registered: Diskret, da nur abzählbar viele Werte angenommen werden können.
- Within San Francisco County: Diskret, da nur abzählbar viele Werte angenommen werden können.

Welchem Skalenniveau entsprechen die einzelnen Merkmale (Nominal-, Ordinal- oder Metrische Skala)?
- Patron Type Definition: Nominalskala
- Total Checkouts: Metrische Skala
- Total Reneweals: Metrische Skala
- Age Range: Ordinalskala
- Home Library Definition: Nominalskala
- Circulation Active Month: Ordinalskala
- Circulation Active Year: Metrische Skala
- Notice Preference Definition: Nominalskala
- Provided Email Address: Nominalskala
- Year Patron Registered: Metrische Skala
- Within San Francisco County: Nominalskala

Enthält der Datensatz fehlende Werte? Ja, in den Spalten Age Range, Home Library Definition, Circulation Active Month, Circulation Active Year, Notice Preference Definition und Within San Francisco County

In [38]:
fehlende_werte = df.isnull().sum()
print(fehlende_werte)

Patron Type Definition              0
Total Checkouts                     0
Total Renewals                      0
Age Range                         510
Home Library Definition            30
Circulation Active Month        40317
Circulation Active Year         40317
Notice Preference Definition     3315
Provided Email Address              0
Year Patron Registered              0
Within San Francisco County       221
dtype: int64


Handelt es sich um Querschnitts-, Längsschnitss- oder Paneldaten? Es handelt sich um Paneldaten, da Beobachtungen von denselben Einheiten (Bibliotheksnutzer*innen) zu verschiedenen Zeitpunkten erhoben werden. 